src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,490 @@
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public class PowerShellFunctionInvoker : FunctionInvokerBase
+    {
+        private readonly ScriptHost _host;
+        private readonly string _scriptFilePath;
+        private readonly string _functionName;
+
+        private readonly Collection<FunctionBinding> _inputBindings;
+        private readonly Collection<FunctionBinding> _outputBindings;
+        private readonly IMetricsLogger _metrics;
+
+        private string _script;
+        private List<string> _moduleFiles;
+        private Dictionary<string, string> _environmentVariables;
+        private Action _reloadScript;
+
+        private static readonly string[] WatchedFileTypes = { ".ps1", ".psm1", ".dll", ".psd1" };
+
+        internal PowerShellFunctionInvoker(ScriptHost host, FunctionMetadata functionMetadata,
+            Collection<FunctionBinding> inputBindings, Collection<FunctionBinding> outputBindings) : base(host, functionMetadata)
+        {
+            _host = host;
+            _scriptFilePath = functionMetadata.Source;
+            _functionName = functionMetadata.Name;
+            _inputBindings = inputBindings;
+            _outputBindings = outputBindings;
+            _metrics = host.ScriptConfig.HostConfig.GetService<IMetricsLogger>();
+            _environmentVariables = new Dictionary<string, string>();
+            InitializeFileWatcherIfEnabled();
+            _reloadScript = ReloadScript;
+            _reloadScript = _reloadScript.Debounce();
+        }
+
+        public override async Task Invoke(object[] parameters)
+        {
+            object input = parameters[0];
+            TraceWriter traceWriter = (TraceWriter)parameters[1];
+            IBinderEx binder = (IBinderEx)parameters[2];
+            ExecutionContext functionExecutionContext = (ExecutionContext)parameters[3];
+            string invocationId = functionExecutionContext.InvocationId.ToString();
+
+            FunctionStartedEvent startedEvent = new FunctionStartedEvent(functionExecutionContext.InvocationId, Metadata);
+            _metrics.BeginEvent(startedEvent);
+
+            try
+            {
+                object convertedInput = input;
+                if (input != null)
+                {
+                    HttpRequestMessage request = input as HttpRequestMessage;
+                    if (request != null)
+                    {
+                        // TODO: Handle other content types? (E.g. byte[])
+                        if (request.Content != null && request.Content.Headers.ContentLength > 0)
+                        {
+                            convertedInput = ((HttpRequestMessage)input).Content.ReadAsStringAsync().Result;
+                        }
+                    }
+                }
+
+                TraceWriter.Verbose(string.Format("PowerShell Function started (Id={0})", invocationId));
```

Comment:
Since you started this work, I overhauled the default verbosity level - it is now Info, not Verbose. See the other script provider. These should be Info level logs, otherwise they won't show up by default.